tests/unit/test_xgboost_handler.py

```diff
@@ -0,0 +1,96 @@
+import os
+import time
+from unittest.mock import patch
+import pandas as pd
+
+from mindsdb_sql import parse_sql
+
+# How to run:
+#  env PYTHONPATH=./ pytest tests/unit/test_xgboost_handler.py
+# Warning: uncomment and run too many sql-testcase in detector and forecaster will cost a long time
+from mindsdb.integrations.handlers.merlion_handler.adapters import DefaultForecasterAdapter, SarimaForecasterAdapter, \
+    ProphetForecasterAdapter, MSESForecasterAdapter, IsolationForestDetectorAdapter, \
+    WindStatsDetectorAdapter, ProphetDetectorAdapter
+# from mindsdb.integrations.handlers.merlion_handler.adapters import DefaultDetectorAdapter
+from .executor_test_base import BaseExecutorTest
+
+import pandas as pd
+from pathlib import Path
+import requests
+
+
+class TestXgboost(BaseExecutorTest):
+    def get_titanic_train_df(self) -> pd.DataFrame:
+        df = pd.read_csv("/home/drill/Data/titanic/train.csv")
```

Comment:
Please change the .csv path to one inside the actual repo (upload a small file if necessary), or maybe to one of our publicly hosted datasets like "Home Rentals"